File: app/src/main/java/com/example/zhibek_romanbekova_hw33_3m/MainFragment.java
package com.example.zhibek_romanbekova_hw33_3m;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

public class MainFragment extends Fragment {
    private RecyclerView rv_flowers;
    private ArrayList<String> flowersList = new ArrayList<>();


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        rv_flowers = requireActivity().findViewById(R.id.rv_flowers);
        loadData();
        initAdapter();
    }
    private void loadData() {
        flowersList.add("Rose");
        flowersList.add("Lily");
        flowersList.add("Popy");
        flowersList.add("Aster");
        flowersList.add("Gladiolus");
        flowersList.add("Iris");
        flowersList.add("Sunflower");
        flowersList.add("Azalea");
        flowersList.add("Cornflower");
        flowersList.add("Hydrangea");
        flowersList.add("Camellia");
        flowersList.add("Pansy");
        flowersList.add("Marigold");
        flowersList.add("Bluebell");
    }

    private void initAdapter() {
        FlowersAdapter adapter = new FlowersAdapter(flowersList);
        rv_flowers.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/zhibek_romanbekova_hw33_3m/MainFragment.kt
package com.example.zhibek_romanbekova_hw33_3m

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.zhibek_romanbekova_hw33_3m.databinding.FragmentMainBinding

class MainFragment : Fragment() {
    private lateinit var binding: FragmentMainBinding
    private var flowersList = ArrayList<Flowers>()
    private val adapter = FlowersAdapter(flowersList)
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentMainBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        loadData()
        initAdapter()
    }

    private fun loadData() {
        flowersList = ArrayList()
        flowersList.add(Flowers("Rose"))
        flowersList.add(Flowers("Lily"))
        flowersList.add(Flowers("Popy"))
        flowersList.add(Flowers("Aster"))
        flowersList.add(Flowers("Gladiolus"))
        flowersList.add(Flowers("Iris"))
        flowersList.add(Flowers("Sunflower"))
        flowersList.add(Flowers("Azalea"))
        flowersList.add(Flowers("Cornflower"))
        flowersList.add(Flowers("Hydrangea"))
        flowersList.add(Flowers("Camellia"))
        flowersList.add(Flowers("Pansy"))
        flowersList.add(Flowers("Marigold"))
        flowersList.add(Flowers("Bluebell"))
    }

    private fun initAdapter() {
        binding.rvFlowers.adapter = adapter
    }
}